### PR TITLE
Enforce server-side that at least one case must be selected

### LINKED ISSUE #268: Do not allow a volunteer without an assigned case to submit a case contact
**What type of user is this for? volunteer/supervisor/admin/all**
unknown

**Description**
When logged in as a user with no assigned case, it is possible to visit the `/case_contacts/new` page and submit an empty form, which results in an error because casa_case_id is nil. 

We *do* want to allow a volunteer without a case to view this page, so that they can see what they will do when they have a case but we don't want the form to be submittable without a case_id specified, so that they don't feel that the system is broken or unreliable. 

**Desired fix:** 
If there is no case associated with the user who is visiting the `/case_contacts/new` page, show a red error saying "no case available"

Since in this case this page was reached by changing the url directly, fixing this bug does not include hiding any "Create case contact" buttons/links on the UI.

**Screenshots of current behavior, if any**
Screen Shot 2020-06-14 at 1 52 27 PM
Screen Shot 2020-06-14 at 1 52 22 PM
Screen Shot 2020-06-14 at 1 52 11 PM

### PR DESCRIPTION
**Pull request by alindeman**
### What github issue is this PR for, if any?
Resolves #268

### What changed, and why?
There's already client-side validation to make sure a user selects at least one case, but this should also be present server-side in case a browser doesn't enforce it properly.

Additionally, if a user is not assigned any cases at all, there will be no checkboxes to select, so the client-side validation may be inadvertently bypassed.

### How will this affect user permissions?
- Volunteer permissions: N/A
- Supervisor permissions: N/A
- Admin permissions: N/A

### How is this tested? (please write tests!) 💖💪

Via local development environment and a new request spec.

I waffled on introducing a higher level feature spec, but decided against it because it seemed like an edge case. I'd be happy to take in feedback about this decision though.

### Screenshots please :)
Screenshot 2020-07-26 at 21 54 53

### CODE REVIEW

File: app/controllers/case_contacts_controller.rb [hunk 1]
```diff
@@ -25,17 +25,27 @@ def new
   end
 
   def create
-    @casa_cases = policy_scope(CasaCase).where(id: params.dig(:case_contact, :casa_case_id))
+    # These variables are used to re-render the form (render :new) if there are
```
Comment (compwron) [inline app/controllers/case_contacts_controller.rb:28]: Great comment! :)

File: spec/requests/case_contacts_spec.rb [hunk 2]
```diff
@@ -63,6 +63,19 @@
         expect(response).to be_successful
       end
     end
+
+    context "with no cases selected" do
+      it "presents the user with a relevant error message" do
```
Comment (compwron) [inline spec/requests/case_contacts_spec.rb:68]: Yay tests! Nice test :)
